core/src/components/button/button.scss

```diff
@@ -210,6 +211,15 @@
 }
 
 
+// Button Slots
+// --------------------------------------------------
+
+::slotted([slot=start]),
+::slotted([slot=end]) {
+  flex-shrink: 0;
+}
```

Comment:
Optional: Looks like this might also resolve https://github.com/ionic-team/ionic-framework/issues/27578? If so, it may be worth backporting the fix, or at least making a note about it on the ticket.

Reply:
I looked into #27578 and unfortunately this won't fix it because [toast uses a native `button` element](https://github.com/ionic-team/ionic-framework/blob/70e7faf6f6c010f3c59787f4cb138fb5664a141d/core/src/components/toast/toast.tsx#L408-L425) instead of an `ion-button`, but the fix should be similar.